tests/api/auth_test.py

```diff
@@ -0,0 +1,57 @@
+import pytest
+
+from zino.api.auth import AuthenticationFailure, authenticate, get_challenge, read_users
+
+
+class TestAuthenticate:
+    def test_when_challenge_response_is_ok_should_return_true(self, secrets_file):
+        assert authenticate(
+            "user1",
+            response="b3e9ba156949152c5c6f3334cf86333b1daabfe9",
+            challenge="a97fbebb3eef2dfcf167645229c0c5fa9e92e3da",
+            secrets_file=secrets_file,
+        )
+
+    def test_when_challenge_response_is_bad_should_raise_error(self, secrets_file):
+        with pytest.raises(AuthenticationFailure):
+            assert authenticate(
+                "user1",
+                response="b3e9ba156949152c5a6f3334df86333b1daabfe9",
+                challenge="a97fbebb3eef2dfcf167645229c0c5fa9e92e3da",
+                secrets_file=secrets_file,
+            )
```

Comment:
These two look identical to me, why should one fail and the other succeed?

Reply:
Ah, well, `b3e9ba156949152c5c6f3334cf86333b1daabfe9` != `b3e9ba156949152c5a6f3334df86333b1daabfe9`.  Do we need them to be obviously different to the human eye? :)

Reply:
alright, new version pushed